tests/test_blueprint_group.py

```diff
@@ -116,6 +116,16 @@ def blueprint_group_1_middleware_not_called(request):
         global MIDDLEWARE_INVOKE_COUNTER
         MIDDLEWARE_INVOKE_COUNTER["request"] += 1
 
+    @blueprint_group_1.on_request
+    def blueprint_group_1_convenience_1(request):
+        global MIDDLEWARE_INVOKE_COUNTER
+        MIDDLEWARE_INVOKE_COUNTER["request"] += 1
```

Comment:
I don't think you need `global` in these because the variable is not being assigned to but rather used via `__setitem__` operator.